src/package-managers/yarn.ts

```diff
@@ -31,15 +31,32 @@ interface YarnConfig {
 
 const TIME_FIELDS = ['modified', 'created']
 
+/** Safely interpolates a string as a template string. */
+const interpolate = (s: string, data: any) => s.replace(/\$\{([^:-]+)(?:(:)?-([^}]*))?\}/g, (match, key, name, fallbackOnEmpty, fallback) => data[key] || (fallbackOnEmpty ? fallback : ''))
+
+/** Reads an auth token from a yarn config, interpolates it, and sets it on the npm config. */
+export const setNpmAuthToken = (npmConfig: Index<string|boolean>, [dep, scopedConfig]: [string, NpmScope]) => {
+  if (scopedConfig.npmAuthToken) {
+    // get registry server from this config or a previous config (assumes setNpmRegistry has already been called on all npm scopes)
+    const registryServer = scopedConfig.npmRegistryServer || npmConfig[`@${dep}:registry`] as string | undefined
+    // interpolate environment variable fallback
+    // https://yarnpkg.com/configuration/yarnrc
+    if (registryServer) {
+      let trimmedRegistryServer = registryServer.replace(/^https?:/, '')
+
+      if (trimmedRegistryServer.endsWith('/')) {
+        trimmedRegistryServer = trimmedRegistryServer.substring(0, trimmedRegistryServer.length - 1)
```

Comment:
Prefer `trimmedRegistryServer.slice(0, -1)`

Reply:
Nice, did not know that one!